### DISCUSSION TITLE
Is there a way we can use force layout along with the expand collapse feature?

### DISCUSSION THREAD

**Discussion by poornima-taranath** [xyflow/xyflow#3086] [General]
I can see that the expand collapse internally uses d3-hierarchy and  the implementation shows tree width, height as params, hence only a tree layout would be possible. Can you guys confirm on this?

I would like to have the initialElements in the expand collapse manner but it should behave/look like force layout. Will this be a feature request (if asked) to your team or is it something that I can only customize ?

Thanks,
Poornima

**Comment by chrtze** [comment]
Hey! I think this should be possible. The expand and collapse example only uses the tree layout for demo purposes but this can be replaced with a different layout algorithm. As this deals with the pro examples which is only intended for subscribers, please reach out via mail (christopher@reactflow.dev) if you need more information.
